src/topology/category/Top/limits.lean

```diff
@@ -195,3 +195,61 @@ end
 end topological_konig
 
 end Top
+
+section fintype_konig
+
+/-- This bootstraps `nonempty_sections_of_fintype_inverse_system`. In this version,
+the `F` functor is between categories of the same universe. -/
+lemma nonempty_sections_of_fintype_inverse_system.init
+  {J : Type u} [directed_order J] (F : Jᵒᵖ ⥤ Type u)
+  [hf : Π (j : Jᵒᵖ), fintype (F.obj j)] [hne : Π (j : Jᵒᵖ), nonempty (F.obj j)] :
+  F.sections.nonempty :=
+begin
+  let F' : Jᵒᵖ ⥤ Top := F ⋙ Top.discrete,
+  haveI : Π (j : Jᵒᵖ), fintype (F'.obj j) := hf,
+  haveI : Π (j : Jᵒᵖ), nonempty (F'.obj j) := hne,
+  obtain ⟨⟨u, hu⟩⟩ := Top.nonempty_limit_cone_of_compact_t2_inverse_system F',
+  exact ⟨u, λ j j' f, hu f⟩,
+end
+
+/-- Gives the induced directed order on the `ulift` of a type with a directed order. -/
+def ulift.directed_order {α : Type u} [directed_order α] : directed_order (ulift.{v} α) :=
+{ le := λ i j, i.down ≤ j.down,
+  le_refl := λ i, le_refl i.down,
+  le_trans := λ i j k hij hjk, le_trans hij hjk,
+  directed := λ i j, begin
+    obtain ⟨k, hk⟩ := directed_order.directed i.down j.down,
+    exact ⟨ulift.up k, hk⟩,
+  end }
+local attribute [instance] ulift.directed_order
+
+/-- The inverse limit of nonempty finite types is nonempty.  This may be regarded
+as a generalization of Kőnig's lemma. -/
+theorem nonempty_sections_of_fintype_inverse_system
+  {J : Type u} [directed_order J] (F : Jᵒᵖ ⥤ Type v)
+  [Π (j : Jᵒᵖ), fintype (F.obj j)] [Π (j : Jᵒᵖ), nonempty (F.obj j)] :
+  F.sections.nonempty :=
+begin
+  let J' := ulift.{v} J,
+  letI : small_category J' := by apply_instance,
+  let jj : J' ⥤ J :=
```

Comment:
I'm not at a computer now, but this must already exist as a general things, and if not it should!

Reply:
Right specifically the equivalence `ulift J` to `J` should be there, and you can just take one direction of it

Reply:
That equivalence apparently doesn't exist yet!

Reply:
What was happening is `ulift` has a category structure from `category_theory.ulift_category`, but there's a second category structure from `preorder.small_category` that uses the `ulift.directed_order` instance from this file.  Since the `ulift` one doesn't create a `small_category`, typeclass resolution found the preorder one!

Since this isn't a good situation, I opted to remove the instance attribute from `ulift.directed_order` and to make the instances explicit.

In the future, the main theorem should be proved for an index category that has a `is_cofiltered` instance, which would make these things a non-issue.  That seems to require this https://stacks.math.columbia.edu/tag/0032 (added a TODO comment for it) when the cofiltered category isn't a thin category.